### PR TITLE
Made views into a package.

### LINKED ISSUE #414: Organize views into a package
### Feature Inspiration

A large `views.py` file is hard to look through and update. 

### Feature Description

Break up `views.py` into multiple modules (and/or packages) in a `views` package.

### Comment

Take care to communicate with other developers so they are aware of this incoming PR and can avoid making large changes to the `views.py` file during this refactor.

-----

### Assumptions

- List of assumptions made WRT the code
- E.g. We will assume input is correct (explaining why there is no validation)

### Requirements

- List of conditions to be met for the feature
- E.g. Every column/row must display a value, i.e. cannot be empty

### Limitations

- A list of things this work will specifically not do
- E.g. This feature will only handle the most frequent use case X

### Affected/Changed Components

- Files
- Environment variables
- External executables
- Database tables
- Cron job settings
- Etc.

#### GUI Change description

Describe changes the user will see.

#### Code Change Description *(Pseudocode optional)*

Describe code changes planned for the feature.

#### Tests

A test should be planned for each requirement (above), where possible.

### PR DESCRIPTION
**Pull request by hepcat72**
#### Deleted
- DataRepo/views.py

#### Modified
- DataRepo/urls.py

#### Added
- DataRepo/views/__init__.py
- DataRepo/views/loading/submission.py
- DataRepo/views/loading/validation.py
- DataRepo/views/models/__init__.py
- DataRepo/views/models/animal.py
- DataRepo/views/models/compound.py
- DataRepo/views/models/msrun.py
- DataRepo/views/models/peakdata.py
- DataRepo/views/models/peakgroup.py
- DataRepo/views/models/peakgroupset.py
- DataRepo/views/models/protocol.py
- DataRepo/views/models/sample.py
- DataRepo/views/models/study.py
- DataRepo/views/models/tissue.py
- DataRepo/views/nav.py
- DataRepo/views/search/advanced/download.py
- DataRepo/views/search/advanced/view.py
- DataRepo/views/search/basic.py
- DataRepo/views/search/results.py
- DataRepo/views/utils.py

## Affected Issue Numbers

- Resolves #414

## Code Review Notes

I changed the example advanced results view name due to prior stated confusion over its purpose.

Views tightly bound to models are in a models subdirectory.

I debated where and how to house the function-based views.  I decided to mix them in and separate things categorically.  So there are no class files like I did with the compositeviews package.

One file name I'm not satisfied with is `DataRepo/views/search/advanced/view.py`.  It's generic, but I did that to not have it be redundant with the parent dir name.

I wanted to pull in some of the data retrieval methods, but I didn't want to create conflicts, so I didn't expand outside of views other than to modify that one view function name.

## Checklist

- [x] All issue requirements satisfied (or no linked issues)
- [x] [Linting passes](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting).
- [x] [Migrations created & committed *(or no model changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)
- [x] [Tests implemented *(or no code changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
- [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)

### CODE REVIEW

**Comment by hepcat72** [comment]
The tests seemed to be stuck.  I manually canceled them at 25 minutes in. I can run them locally in somewhere under 8 minutes.  Going to see if I can restart them manually.

**Comment by hepcat72** [comment]
Well, now the job (started) is just sitting and waiting for a runner.  I suspect something is going on on the github side...

**Comment by hepcat72** [comment]
Huh.  Wow.  That was fast...

**Review by lparsons — APPROVED** [review]
Looks like a logical organization to me, really breaks things up nicely. If you wanted a suggestion about naming the advanced search stuff, my one thought for an option would be:
```
search/
  advanced.py
  basic.py
  download.py  
  results.py
  utils.py
```

Similar to the other refactors of this nature, I'm assuming that the code didn't really change, just the organization so I focused my review on the organization and not the specific code.

**Review by fkang-pu — APPROVED** [review]
Looks good to me!

**Review by jcmatese — APPROVED** [review]
Seem fine to me.  The questions I have are less to do with this specific PR, but perhaps the longterm project implications/maintenance/rules.

- For "nested" packages (which include `__init__.py` content like `views/` and `views/models/` ), if one were to to add a new model view, do you have to touch both `__init__.py` files? or perhaps this will be decided as we go?
- Some of these new directories don't appear to have `__init__.py` files (`/views/loading/` `/views/search/`, `/views/search/advanced` [_at least looking at the left panel GitHub tree_]).  Is that by design? Is there a rule for when to include a `__init__.py` file, and when not?

File: DataRepo/urls.py [hunk 1]
```diff
@@ -14,9 +14,9 @@
     ),
     # Remove this when done
     path(
-        "test_barebones_advanced_search/",
-        views.test_barebones_advanced_search,
-        name="test_barebones_advanced_search",
+        "example_barebones_advanced_search/",
+        views.example_barebones_advanced_search,
+        name="example_barebones_advanced_search",
     ),
```
[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/urls.py:20]: I think by "done", I was referring to your expressed desire (in 1 or 2 meetings a couple months ago or so) to link results to the advanced search by providing only a queryset.  At least, I thought about that when I touched this code yesterday and explicitly decided to keep it until the example is not longer necessary/useful.  If you think you can accomplish that without this example as a guide, then I'm fine with removing it.

[the remark this answers is not in the document]
Comment (lparsons) [reply DataRepo/urls.py:20]: No need to keep it on my account.

**Comment by hepcat72** [comment]
> Seem fine to me. The questions I have are less to do with this specific PR, but perhaps the longterm project implications/maintenance/rules.
> 
> * For "nested" packages (which include `__init__.py` content like `views/` and `views/models/` ), if one were to to add a new model view, do you have to touch both `__init__.py` files? or perhaps this will be decided as we go?

I had the same exact question yesterday when I was using the models example as a guide.  I'd googled it and found [this](https://stackoverflow.com/questions/34855071/importing-all-functions-from-a-package-from-import).  I'd thought to comment on the answer that says that - all imports must always be explicit and that, yes, you have to touch `__init__.py` every time - to ask if there existed an analog to perl's `EXPORT` and `EXPORT_OK` variables.  I decided I already knew the pythonic answer, so I didn't ask.

So the answer, if we stick to python conventions, is that we do have to touch both `__init__.py` files.

> * Some of these new directories don't appear to have `__init__.py` files (`/views/loading/` `/views/search/`, `/views/search/advanced` [_at least looking at the left panel GitHub tree_]).  Is that by design? Is there a rule for when to include a `__init__.py` file, and when not?

That's a valid question.  I think I was just trying to make imports more straightforward for files that import the contained methods/classes.  But I'm open to discussing alternatives.

File: DataRepo/urls.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/urls.py:20]: - [ ] Remove the advanced search view mode example

**Comment by lparsons** [comment]
> > Seem fine to me. The questions I have are less to do with this specific PR, but perhaps the longterm project implications/maintenance/rules.
> > 
> > * For "nested" packages (which include `__init__.py` content like `views/` and `views/models/` ), if one were to to add a new model view, do you have to touch both `__init__.py` files? or perhaps this will be decided as we go?
> 
> I had the same exact question yesterday when I was using the models example as a guide. I'd googled it and found [this](https://stackoverflow.com/questions/34855071/importing-all-functions-from-a-package-from-import). I'd thought to comment on the answer that says that - all imports must always be explicit and that, yes, you have to touch `__init__.py` every time - to ask if there existed an analog to perl's `EXPORT` and `EXPORT_OK` variables. I decided I already knew the pythonic answer, so I didn't ask.
> 
> So the answer, if we stick to python conventions, is that we do have to touch both `__init__.py` files.

In my experience, including such imports in `__init__.py` is rare. The only time I'd ever done it was in the `models.py` refactor mostly as a proof of concept. It had the side effect that we didn't have to make changes to imports, but given the maintenance headaces, that is a tradeoff.

> > * Some of these new directories don't appear to have `__init__.py` files (`/views/loading/` `/views/search/`, `/views/search/advanced` [_at least looking at the left panel GitHub tree_]).  Is that by design? Is there a rule for when to include a `__init__.py` file, and when not?
> 
> That's a valid question. I think I was just trying to make imports more straightforward for files that import the contained methods/classes. But I'm open to discussing alternatives.

Good catch John. I believe that all directories should have an `__init__.py` file (all directories that are intended to be packages). See [Python Package docs](https://docs.python.org/3/tutorial/modules.html#packages).

File: DataRepo/urls.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/urls.py:20]: I actually decided to make it a generic view function that takes a queryset and a format.

**Comment by hepcat72** [comment]
Should have mentioned in the log that I also added the requested init files.

**Pull request merged**